Adaptors/Amqp/src/ServiceCollectionExt.cs

```diff
@@ -105,9 +105,12 @@ public static class ServiceCollectionExt
       }
 
       var sessionProvider = new SessionProvider(amqpOptions,
-                                                logger);
+                                              logger);
 
-      serviceCollection.AddSingleton(sessionProvider);
+      serviceCollection.AddSingleton<IProviderBase<Session>, SessionProvider>(sp =>
+        {
```

Comment:
Couldn't you just write: `sp => sessionProvider` without having a block?

Reply:
Done